### PR TITLE
support for packaging in multiple cuda runners

### PR DESCRIPTION
**Pull request by BruceMacD**

### CODE REVIEW

File: llm/llama.cpp/generate_linux.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+# only do GPU builds if nvcc is installed
+if which nvcc > /dev/null; then
```
Comment (BruceMacD) [inline llm/llama.cpp/generate_linux.sh:3]: Looking for feedback on how to get the CUDA version for the filepath in go generate, I was having trouble with this. Any alternative suggestions welcome also.

File: llm/llama.cpp/generate_linux.go [hunk 2]
```diff
@@ -6,9 +6,11 @@ package llm
 //go:generate -command git-apply git -C ggml apply
 //go:generate git-apply ../ggml_patch/0001-add-detokenize-endpoint.patch
 //go:generate git-apply ../ggml_patch/0002-34B-model-support.patch
-//go:generate cmake -S ggml -B ggml/build/gpu -DLLAMA_CUBLAS=on -DLLAMA_ACCELERATE=on -DLLAMA_K_QUANTS=on 
-//go:generate cmake --build ggml/build/gpu --target server --config Release
+//go:generate cmake -S ggml -B ggml/build/cpu -DLLAMA_K_QUANTS=on
```
Comment (BruceMacD) [inline llm/llama.cpp/generate_linux.go:9]: without the cpu generation here the cpu runner was not packaged in my tests

File: llm/llama.cpp/generate_linux.sh
Comment (mxyng) [inline llm/llama.cpp/generate_linux.sh:1]: I'm not a big fan of calling a script inside of `go:generate`. It seems to me the only reason so the cuda version can be parsed. Perhaps a simpler way is to pass in the version through environment variables?

```go
//go:generate cmake -S gguf/gguf/build/gpu-${CUDA_VERSION} ...
```

Comment (BruceMacD) [reply llm/llama.cpp/generate_linux.sh:1]: This is simpler, thanks for the suggestion. I went with this env var approach.

**Review by mxyng — APPROVED** [review]

**Pull request merged**